### ISSUE TITLE
Long runtime on A100 GPU 

### ISSUE THREAD

**Issue by matsuobasho** [samlhuillier/code-llama-fine-tune-notebook#5]
Thanks so much for this notebook, it's really helpful and educational.  You mentioned how finetuning took you about an hour on an A100 running Python 3.10 and Cuda 11.8.

I'm running an environment with Python 3.8 and Cuda 11.7, so maybe that's the issue, but I've been running for ~45 minutes and still have only 0.14 of an epoch completed:

{'loss': 0.5827, 'learning_rate': 0.000237, 'epoch': 0.14} 

Any thoughts on whether it's the older Python and Cuda versions that may be causing this long run-time or maybe something else?
